txn.go

```diff
@@ -147,7 +147,11 @@ func (txn *Txn) Insert(table string, obj interface{}) error {
 
 	// Get the primary ID of the object
 	idSchema := tableSchema.Indexes[id]
-	ok, idVal, err := idSchema.Indexer.FromObject(obj)
+	idIndexer, ok := idSchema.Indexer.(SingleIndexer)
+	if !ok {
```

Comment:
I would push this check into the `Validate` of the schema, just so it can be avoided here